wrapanapi/systems/ec2.py

```diff
@@ -804,17 +807,25 @@ def object_exists_in_bucket(self, bucket_name, object_key):
         objects = [o for o in bucket.objects.all() if o.key == object_key]
         return any(objects)
 
-    def delete_s3_bucket(self, bucket_name):
-        """TODO: Force delete - delete all objects and then bucket"""
-        bucket = self.s3_connection.Bucket(bucket_name)
-        self.logger.info("Trying to delete bucket '%s'", bucket_name)
-        try:
-            bucket.delete()
-            self.logger.info("Success: bucket '%s' was deleted.", bucket_name)
-            return True
-        except Exception:
-            self.logger.exception("Bucket '%s' deletion failed", bucket_name)
-            return False
+    def delete_s3_buckets(self, bucket_names):
+        """ Deletes specified bucket(s) with keys """
+        deleted_list = []
+        if isinstance(bucket_names, (set, list, tuple)):
+            buckets = [self.s3_connection.Bucket(obj_name) for obj_name in bucket_names]
+        else:
+            raise ValueError("Object is not iterable.")
+        for bucket in buckets:
+            self.logger.info("Trying to delete bucket '%s'", bucket)
```

Comment:
QUESTION: is bucket correctly displayed in logs ? 
Shouldn't you pass bucket name instead ?

Reply:
@izapolsk the bucket contains bucket name only.